external-crates/move/crates/move-compiler/tests/move_check_testsuite.rs

```diff
@@ -273,7 +273,14 @@ fn out_path(path: &Path, test_name: &str, test_kind: Option<&str>) -> PathBuf {
 // Runs all tests under the test/testsuite directory.
 pub fn run_test(path: &Path) -> datatest_stable::Result<()> {
     let (test_kind, test_info, package_config, flags) = test_config(path);
-    let test_name: &str = path.file_name().unwrap().to_string_lossy().as_ref();
+    let migration_mode = package_config.edition == Edition::E2024_MIGRATION;
+    let p;
+    let test_name = {
+        p = path.with_extension("");
+        assert!(p.extension().is_none());
+        p.file_name().unwrap().to_string_lossy()
+    };
```

Comment:
I believe this is equivalent to
```rust
let test_name = path.file_stem().unwrap().to_string_lossy().as_ref();
```
and that might be a bit cleaner?

Reply:
Thanks! I was looking for this!